### ISSUE TITLE
Fashion Attribute-to-Image Synthesis Using Attention-based Generative Adversarial Network

### ISSUE THREAD

**Issue by reyllama** [reyllama/paper-reviews#167]
## TL; DR

- Trains an attribute-conditioned GAN in fashion domain with product-of-Gaussian generator latent and attentional discriminator.

## Motivation

- Conventional conditional GANs usually deal with few class labels unlike in fashion domain where a wide variety of attributes can be provided as conditioning information.

## Formulation

<img width="1224" alt="image" src="https://user-images.githubusercontent.com/46308407/162152320-79177fd3-daea-40b2-b0a9-c43fa4509faf.png">

### Product-of-Gaussian Generator Latent Prior

- *Compositional abstraction hierarchy*: conditionally generated samples should cover unobserved attributes
- Each *attribute embedding vector* is mapped to Gaussian parameter (*mean and variance*), and then combined as follows:
<img width="650" alt="image" src="https://user-images.githubusercontent.com/46308407/162152715-13df4c84-7d4c-4e48-935e-33c30d4b4efd.png">

### Attribute-attentional Discriminator

- Intermediate discriminator features take the form **D x H**, where **D** refers to feature dimension and **H** represents the number of subregions (*how did they get It tho..?*)
- Each attribute vector is projected to *D*-dimensional feature space using learnable embedding, and then attention score for each subregion is calculated as following:
<img width="297" alt="image" src="https://user-images.githubusercontent.com/46308407/162153405-03e26fef-394f-4337-ad8a-6327b8a63515.png">
- Here, *K* is the number of attributes in the set.
- The coefficients are learned to represent how much each attribute is related to synthesizing the given subregion.
- The context vector for the subregion is first aggregated and concatenated to the original discriminator feature.
<img width="206" alt="image" src="https://user-images.githubusercontent.com/46308407/162153688-2a5920f9-e3ee-4af2-a6e8-2de8ef93c2d4.png">

### Visual-Semantic Embedding

- Since fashion attributes are domain-specific, naive use of word embeddings can result in overly coarse embedding representation. 
- Given a pair of [image, attribute set], learn a convolutional image encoder and attribute embedding.
- Since there can be more than one attribute for an image, attribute embeddings are simply averaged.
- Averaged attribute embedding and convolutional image features are enforced by triplet metric learning objective.

<img width="468" alt="image" src="https://user-images.githubusercontent.com/46308407/162154329-28efb9e5-764c-46c6-bfd3-14af27dbce68.png">

### Dataset

- Collect over 30,000 images of dress and corresponding product title from *polyvore.com*.
- To obtain fashion attributes, tokenize the title by 6 criteria: *length, sleeve length, neckline, color, pattern, style*. (*kudos* 👍 )

<img width="1169" alt="image" src="https://user-images.githubusercontent.com/46308407/162154408-fe11d899-afa1-4d95-b4fd-6a33c6d3b881.png">
<img width="727" alt="image" src="https://user-images.githubusercontent.com/46308407/162154435-5456bbbe-b6f5-49e4-b163-6a690d65de16.png">

### Metrics

- Although Inception Score (IS) is widely used in evaluation of generative models, the inception network is trained on general images from ImageNet, rendering it inadequate for domain-specific synthesis, which is the case for ADGAN.
- As a detour, the authors train an oracle fashion attribute classifier that shows about 76% accuracy.
- ***Correctness*** measures whether the oracle classifier correctly classifies the synthesized image.
<img width="564" alt="image" src="https://user-images.githubusercontent.com/46308407/162155823-4fc12400-fa59-4640-bd5b-ce7c1d117533.png">

- ***Correctness_human*** is evaluated by 10 people that are not involved in this paper.
- ***Coverage*** measures the JS divergence on unseen attribute frequency distribution as follows:
<img width="528" alt="image" src="https://user-images.githubusercontent.com/46308407/162155888-90dc2967-3df7-44d3-b2f6-92ff93eb538a.png">
- ***MS-SSIM*** is calculated between pairs of generated images to gauge the degree of mode collapse.

### Results

<img width="1461" alt="image" src="https://user-images.githubusercontent.com/46308407/162156121-e8eb1c30-138d-43f5-8b3a-dcbe0bd16f3a.png">
<img width="1102" alt="image" src="https://user-images.githubusercontent.com/46308407/162156138-45245143-a11a-47d7-b6b1-b82200082bc8.png">
<img width="710" alt="image" src="https://user-images.githubusercontent.com/46308407/162156200-ce1c0ccc-2fcd-4d14-8bd3-e743a5628112.png">
<img width="1465" alt="image" src="https://user-images.githubusercontent.com/46308407/162156238-b43671f5-ea7c-4089-abb7-f8422662a6e8.png">

## Comments

- Great work collecting the dataset. In fashion domain like ViTON, obtaining adequate dataset is a big challenge.
- Since they collected attribute annotation from parsing product title, the set is not extensive enough. This more or less contradicts their motivation of the problem: fashion attributes are very diverse and thus inappropriate to model with normal conditional generators.
- As for the method, there is quite nothing that stands out, but its adequate overall to address the proposed challenges.
